src/Engine/GameEvents/TurnInitEvent.java

```diff
@@ -47,12 +50,14 @@ public GameAnimation getEventAnimation(MapView mapView)
   @Override
   public GameEventQueue sendToListener(GameEventListener listener)
   {
-    return listener.receiveTurnInitEvent(army, turn);
+    return listener.receiveTurnInitEvent(map, army, turn);
   }
 
   @Override
-  public void performEvent(MapMaster gameMap)
+  public void performEvent(MapMaster map)
   {
+    for( Unit u : army.getUnits() )
+      u.initTurn(map);
```

Comment:
This appears to fire for every unit, every turn. We previously had a caveat for off-map units (though we did init cargo units - I don't recall why, or how else a unit could be at an invalid location).

Reply:
We didn't at time of writing.
In fact, cargo was init'd *twice* - once by `Commander`, once by its holder.

As for "Why init cargo?", `LAUNCH` depends on whether that unit has acted this turn or not.

Reply:
> LAUNCH

Right, fair.

I'm still seeing this get hit for each army, each turn, which is strange.

Reply:
I would expect only one `TurnInit` to go through each turn, so yeah, that's pretty weird.

Reply:
Looks like the `TurnInitEvent` is hitting every unit, every turn, but `Commander` only hits `UnitModel.turnInitEvent` on his own turn. Meaning that repairs happen correctly, but fuel burn is effectively multiplied by the number of players.

Reply:
Looks like `TurnEndEvent` inherits from `TurnInitEvent` but doesn't override everything.

Reply:
Whoops. Thanks.